pytorch_lightning/callbacks/pt_callbacks.py

```diff
@@ -228,30 +236,34 @@ def on_epoch_end(self, epoch, logs=None):
         self.epochs_since_last_save += 1
         if self.epochs_since_last_save >= self.period:
             self.epochs_since_last_save = 0
-            filepath = '{}/{}_ckpt_epoch_{}.ckpt'.format(self.filepath, self.prefix, epoch + 1)
-            if self.save_best_only:
+            if self.save_all or self.save_last:
+                overwrite = not self.save_all
+                filepath = '{}/{}_ckpt_epoch_{}.ckpt'.format(self.filepath, self.prefix, epoch + 1)
+                self.save_model(filepath, overwrite=overwrite, prefix=self.prefix)
```

Comment:
why is the prefix used twice - here and also in the file name?

Reply:
`ModelCheckpoint.save_model` had a bug (#394) where the whole directory is wiped. Passing `self.prefix` to the method allows me to then identify the files that need deletion by using re (`re.match(r'{}_ckpt_epoch_\d+.ckpt'.format(prefix), filename)`)